process/block/metablock.go

```diff
@@ -1749,3 +1754,45 @@ func (mp *metaProcessor) getPendingMiniBlocks() []bootstrapStorage.PendingMiniBl
 
 	return pendingMiniBlocks
 }
+
+func (mp *metaProcessor) generateValidatorMiniBlocks() (block.Body, error) {
+	validatorStatsRootHash, err := mp.validatorStatisticsProcessor.RootHash()
+	if err != nil {
+		return nil, err
+	}
+
+	validatorInfoDataList, err := mp.validatorStatisticsProcessor.GetValidatorInfoForRootHash(validatorStatsRootHash)
+	if err != nil {
+		return nil, err
+	}
+
+	return mp.createValidatorMiniBlocks(validatorInfoDataList)
+}
+
+func (mp *metaProcessor) createValidatorMiniBlocks(list map[uint32][]*state.ValidatorInfoData) ([]*block.MiniBlock, error) {
```

Comment:
please look as how end of epoch rewards and economics were done, and make a small interface for this creation, verification is needed as well at process block.